tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,500 @@
+            String[] coordinatesParts = answer.split(":");
+            if (coordinatesParts.length != 3) {
+                throw new IllegalStateException("Maven coordinates not provided in the correct format. Type help for explanation.");
+            }
+
+            String group = coordinatesParts[0];
+            String artifact = coordinatesParts[1];
+            String version = coordinatesParts[2];
+            return new Coordinates(group, artifact, version);
+        });
+    }
+
+    private Path getTestsLocation() {
+        ContributingQuestion question = questions.get("testsLocation");
+        return InteractiveTaskUtils.askQuestion(question.question(), question.help(), (answer) -> {
+            Path testsLocation = Path.of(answer).toAbsolutePath();
+            if (!Files.exists(testsLocation)) {
+                throw new IllegalStateException("Cannot find tests directory on the given location: " + testsLocation + ". Type help for explanation.");
+            }
+
+            if (!Files.isDirectory(testsLocation)) {
+                throw new IllegalStateException("Provided path does not represent a directory: " + testsLocation + ". Type help for explanation.");
+            }
+
+            checkPackages(testsLocation);
+
+            return testsLocation;
+        });
+    }
+
+    private void checkPackages(Path testsPath) {
+        List<Path> javaFiles = new ArrayList<>();
+        try {
+            FilesUtils.findJavaFiles(testsPath, javaFiles);
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot find java files. Reason: " + e);
+        }
+
+        javaFiles.forEach(file -> {
+            try {
+                Optional<String> packageLine = Files.readAllLines(file).stream().filter(line -> line.contains("package ")).findFirst();
+                if (packageLine.isEmpty()) {
+                    throw new RuntimeException("Java file: " + file + " does not contain declared package");
+                }
+
+                String declaredPackage = packageLine.get().split(" ")[1].replace(";", "");
+                String packagePath = declaredPackage.replace(".", File.separator);
+                if (!Files.exists(testsPath.resolve(packagePath))) {
+                    throw new IllegalStateException("File: " + file + " has package: " + declaredPackage +
+                            " that cannot be found on tests location: " + testsPath +
+                            ". Please make sure that the location you provided is a directory that contains packages with tests implementations");
+                }
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    private Path getResourcesLocation(){
+        ContributingQuestion question = questions.get("resourcesLocation");
+        return InteractiveTaskUtils.askQuestion(question.question(), question.help(), (answer) -> {
+            if (answer.equalsIgnoreCase("-")) {
```

Comment:
Wouldn't be more user-friendly to end a sequence of items with just <enter>?

E.g.

```
foo<enter>
bar<enter>
<enter>
```
vs.

```
foo<enter>
bar<enter>
-<enter>
```

Also you could provide a generalized form of asking for a list of items in `InteractiveTaskUtils`

Reply:
> Wouldn't be more user-friendly to end a sequence of items with just
I am slightly against that approach... I intentionally added this required `-` because I want to avoid a situation where user accidentally presses `enter` two times and terminates the process unintentionally. 

> Also you could provide a generalized form of asking for a list of items in InteractiveTaskUtils and use it wherever you need lists of items.
Implemented here: https://github.com/oracle/graalvm-reachability-metadata/pull/605/files#diff-a552aa0289476dfc397600ecadf14028ef3677d3411da87fa24084ab1636db40R36